corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js

```diff
@@ -423,12 +424,27 @@ hqDefine("cloudcare/js/formplayer/menus/views", function () {
         },
 
         selectAllAction: function (e) {
+            this.children.each(function (childView) {
+                let checkbox = childView.ui.selectRow[0];
+                if (checkbox.checked === !e.target.checked) {
+                    checkbox.checked = e.target.checked;
+                }
+            });
             if (e.target.checked) {
-                // add all rows to selected cases
+                for (const value of this.data) {
+                    if (this.selectedCaseIds.indexOf(value.id) === -1) {
+                        this.selectedCaseIds.push(value.id);
+                    }
+                }
             } else {
-                // remove all rows from selected cases
+                for (const value of this.data) {
```

Comment:
@Robert-Costello @orangejenny changed my mind about setting `selectedCaseIds` to`[]` here and rather it removes based on index one at a time. The thought here is that it more matches the logic of the select all and follows more along the lines of "deselect all from page" which is what we are going for

Reply:
Sounds good. Thinking about it, it should also be be better if a user has cases that were selected on another page.